cep-XXXX.md

```diff
@@ -0,0 +1,229 @@
+
+- channel: `^[a-z0-9]+((-|_|.)[a-z0-9]+)*$`
+- subdirs: `^[a-z0-9]+((-|_|.)[a-z0-9]+)*$`
+- label: `^[a-zA-Z][0-9a-zA-Z_\-\.\/:\s]*`
+- package name: `^(([a-z0-9])|([a-z0-9_](?!_)))[._-]?([a-z0-9]+(\.|-|_|$))*$`
+
+All channels, subdirs, labels, and package names MUST conform to their respective regex in the table above.
+
+Further, the following rule applies to labels:
+
+- The label `NOLABEL` is reserved and MUST only be used for conda packages which have no other labels. In other words, in the space of labels, the empty set is represented by the label `NOLABEL`.
+
+### Mapping conda Packages with Channels, Subdirs, Labels to OCI Repositories and Tags
+
+Per the [OCI Distribution Spec](https://github.com/opencontainers/distribution-spec) v1.0, the `<name>` and `<tag>` of an OCI repository are subject to the following regexes:
+
+- `<name>`: `^[a-z0-9]+((\.|_|__|-+)[a-z0-9]+)*(\/[a-z0-9]+((\.|_|__|-+)[a-z0-9]+)*)*$`
+- `<tag>`: `^[a-zA-Z0-9_][a-zA-Z0-9._-]{0,127}$`
+
+The [OCI Distribution Spec](https://github.com/opencontainers/distribution-spec) further notes that "Many clients impose a limit of 255 characters on the length of the concatenation of the registry hostname (and optional port), /, and <name> value."
+
+These constraints motivate the following set of rules for mapping conda packages to OCI repositories and tags. More explanation is given below in the Rational section and the example implementation illustrate how these rules work together to create a human-readable and easily parsable OCI repository `<name>` and `<tag>` for a conda package.
+
+#### Overall Form of the OCI Repository and Tag
+
+A conda package with its channel, subdir, and (optional) label MUST be mapped to an OCI `<repository>:<tag>` in one of two forms.
+
+The unhashed form of the OCI repository and tag for a conda package is:
+
+```
+<channel>/<subdir>/<OCI-encoded package name>:<OCI-encoded version, build and (optional) label in the form '<version>-<build>(-<label>)'>
+```
+
+In the hashed form, the OCI-form of the package name and the string composed of the OCI-forms of the version, build, and (optional) label (i.e., `<version>-<build>(-<label>)`) are each separately hashed via SHA1 and then replaced by the string `h<hexdigest of SHA1 hash>` as follows
+
+```
+<channel>/<subdir>/h<hexdigest of the SHA1 hash of OCI-encoded package name>:h<hexdigest of the SHA1 hash of the original OCI tag of the conda package>
+```
+
+The `<channel>` and `<subdir>` MUST be used as-is without modification. The OCI-forms of the package name, label, version, and build string are defined below.
+
+#### Encoding Package Names, Channels, and Subdirs into OCI Repository `<name>`s
+
+The package name MUST be encoded to OCI-form as follows:
+
+- If the package name starts with an `_`, the `_` MUST be replaced with `z`.
+- Otherwise, if the package name does NOT start with an `_`, the package name MUST be prepended by `c`.
```

Comment:
Can you elaborate on why these characters were chosen? I assume `z` is just to sort last, and `c` for `conda` just to have something else?

Reply:
I was following the suggestion from @baszalmstra.